### PR TITLE
Issue #773: Ensure that bash fixes are always selected

### PR DESCRIPTION
**Pull request by jan-cerny**
Bash is a default language for remediation scripts. If we don't
provide a template explicitely, the result script shouldn't
contain Ansible or other snippets.

### CODE REVIEW

**Comment by mpreisler** [comment]
I am in favor of this. Have you checked why we don't do this already? Maybe there is a reason. Selecting a default explicitly is an obvious solution so maybe there is a catch.

Could we also get some shorthand for --fix_type bash? Instead of having to type the huge urn style template?

**Comment by redhatrises** [comment]
Having the `--fix-type` or `--script-type` would be a handy feature. You could make it so that the default of the option is set to bash.

**Comment by jan-cerny** [comment]
@mpreisler I don't think there is, I think we only haven't noticed before, because there used to be only bash in SSG. There is a code that selects the best fix according to various criteria - reboot, disruption, etc, but those criteria should be considered AFTER we select a language.

I think bash should be default if `--template` isn't specified, because I have found this in the man page:
```
                     --template ID|FILE
                            Template to be used to generate the script. If  it
                            contains a dot '.' it is interpreted as a location
                            of a file with the template definition.  Otherwise
                            it  identifies  a template from standard set which
                            currently includes: bash (default if no --template
                            switch  present). Brief explanation of the process
                            of writing your own templates is in the  XSL  file
                            xsl/legacy-fix.xsl in the openscap data directory.
                            You can also take a look at the  default  template
                            xsl/legacy-fixtpl-bash.xml.
```
and --help says:
```
   --template <id|filename>      - Fix template. (default: bash)
```

@redhatrises Yes, that would be nice. I can't remember `urn:xccdf:fix:script:sh`. I usually write it wrong. I suppose our users don't remember it as well. :) Moreover it's badly documented

**Comment by jan-cerny** [comment]
Btw, `template` sounds weird for this.

**Comment by jan-cerny** [comment]
I have added `--fix-type` option into the "oscap xccdf generate fix" module.

File: utils/oscap-xccdf.c [hunk 1]
```diff
@@ -791,11 +792,41 @@ int app_generate_fix(const struct oscap_action *action)
...
+				"Option '--fix-type' is mutually exclusive with '--template'.\n"
+				"Please provide only one of them.\n");
+		return OSCAP_ERROR;
+	} else if (action->fix_type != NULL) {
+		if (strcmp(action->fix_type, "bash") == 0) {
+			template = "urn:xccdf:fix:script:sh";
+		} else if (strcmp(action->fix_type, "ansible") == 0) {
+			template = "urn:xccdf:fix:script:ansible";
+		} else if (strcmp(action->fix_type, "puppet") == 0) {
+			template = "urn:xccdf:fix:script:puppet";
```
Comment (mpreisler) [inline utils/oscap-xccdf.c]: anaconda would also be nice so that we support everything we use in SSG.

File: utils/oscap.8 [hunk 2]
```diff
@@ -331,6 +331,9 @@ Result-oriented fixes are generated using result-id provided to select only the
 .TP
 Profile-oriented fixes are generated using all rules within the provided profile. If no result-id/profile are provided, (default) profile will be used to generate fixes.
 .TP
+\fB\-\-fix-type TYPE\fR
+Specify fix type. There are multiple programming languages in which the fix script can be generated. TYPE should be one of: bash, ansible, puppet. Default is bash.
```
Comment (mpreisler) [inline utils/oscap.8]: Explain that it is mutually exclusive with --template and why. Add anaconda.

**Comment by mpreisler** [comment]
Fails the test suite. The failure looks legitimate.

File: utils/oscap-xccdf.c (hunk 1 above)
Comment (jan-cerny) [inline utils/oscap-xccdf.c]: OK

File: utils/oscap.8 (hunk 2 above)
Comment (jan-cerny) [inline utils/oscap.8]: OK

**Comment by jan-cerny** [comment]
There is a problem: attribute `system` of element `fix` isn't required.

**Comment by yuumasato** [comment]
LGTM. Ack.

**Pull request merged**